Makefile

```diff
@@ -45,12 +45,13 @@ build-startup-script:
 
 build-test-script:
 	echo '#!/bin/bash' > $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'set -e' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
+	echo 'set -x' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
 	echo 'script_dir=$$(dirname "$$(readlink -f "$$0")")' >> $(TEST_DIR)/$(TEST_SCRIPT_NAME)
```

Comment:
this two lines are added to ask bash to capture the errors and failures in whole test_script.sh, instead of returning the exit code of the last command line in default.